doc/clone-subdomain.md

```diff
@@ -0,0 +1,42 @@
+# Cloning a Subdomain
+
+To connect a new clone to a Gateway subdomain, use the configuration provided in the domain creation response. For Socket.io messaging (the default), the configuration will look like this:
+
+```json
+{
+  "@domain": "my-subdomain.my-account.my-gateway",
+  "genesis": false,
+  "io": {
+    "uri": "http://my-gateway/",
+    "opts": {
+      "auth": {
+        "user": "my-account",
+        "key": "≪your-auth-key≫"
+      }
+    }
+  }
+}
+```
+
+Note that for UUID subdomains, the `genesis` flag defaults to `true`, as the Gateway does not know whether the subdomain already exists.
+
+To use this config, you must augment it as follows:
```

Comment:
**thought:** The fact that this config requires so much massaging makes me suspect that the shape of the config may not currently be ideal.

Reply:
Yah, agreed. Some parochial thoughts, leaving aside the big ideas we discussed:
1. We could generate an `@id`. The risk here is that an app could give out the response config to several clones. Maybe better to have the m-ld `clone` method generate an id if it's not given one.
2. For the same reason I'm not happy reflecting the key used in the config request – or a Bearer token for that matter, also since it will expire.
3. If the domain is 'live' (there are active clones), having a false positive genesis flag is actually safe. The risk is that it's not live and you accidentally create a genesis clone which will immediately terminate when another genesis clone appears.

Hum. Perhaps a good answer here is to separate 'domain' config from 'clone' config. 'Domain' config is what the Gateway gives you, and it's the same for all clones. 'Clone' config is what you need to add per-clone. (The Gateway actually already has this distinction, with the idea of 'sharable' config.)

In the clone config we could also have some default placeholders for common ideas like auth credentials, which can be picked up by specific remotes if applicable. So, no matter whether you're using Socket.io or Ably, you always set `remotes.user` and `remotes.key` or `remotes.token`.

Reply:
> Hum. Perhaps a good answer here is to separate 'domain' config from 'clone' config. 'Domain' config is what the Gateway gives you, and it's the same for all clones. 'Clone' config is what you need to add per-clone. (The Gateway actually already has this distinction, with the idea of 'sharable' config.)

I like this! I also like the idea of the clone generating its own `@id`, at least optionally. There's no meaning to that value other than uniqueness, right? That is, the app that's currently generating it and passes it in never needs it again? That smells to me like the clone should be generating it for itself.

And check it out! Once the domain config is separate from the clone config, `@domain` becomes an `@id`! It's a JSON-LD node object describing the domain! (Especially if the domain's name becomes an IRI.)